code/modules/antagonists/traitor/smart_objectives.dm

```diff
@@ -0,0 +1,323 @@
+	var/datum/techweb/checking = new
+	var/list/datum/mind/owners = get_owners()
+	for(var/datum/mind/owner in owners)
+		if(ismob(owner.current))
+			var/mob/M = owner.current //Yeah if you get morphed and you eat a quantum tech disk with the RD's latest backup good on you soldier.
+			if(ishuman(M))
+				var/mob/living/carbon/human/H = M
+				if(H && (H.stat != DEAD) && istype(H.wear_suit, /obj/item/clothing/suit/space/space_ninja))
+					var/obj/item/clothing/suit/space/space_ninja/S = H.wear_suit
+					S.stored_research.copy_research_to(checking)
+			var/list/otherwise = M.GetAllContents()
+			for(var/obj/item/disk/tech_disk/TD in otherwise)
+				TD.stored_research.copy_research_to(checking)
+	return checking.researched_nodes.len >= target_amount
+
+/datum/objective/smart/download/post_find_target()
+	if(!target || !target.current)
+		return
+	RegisterSignal(SSdcs, COMSIG_GLOB_TECHDISK_EJECTED, .proc/on_techdisk_ejected)
+
+/datum/objective/smart/download/Destroy(force)
+	. = ..()
+	UnregisterSignal(target.current, list(COMSIG_GLOB_TECHDISK_EJECTED))
+
+/datum/objective/smart/download/proc/on_techdisk_ejected(datum/target, obj/item/disk/tech_disk/tech_disk)
+	SIGNAL_HANDLER
+
+	///hey this would complete our objective if we held it
+	if(tech_disk.stored_research.researched_nodes.len >= target_amount)
+		RegisterSignal(tech_disk, COMSIG_ITEM_PICKUP, .proc/on_objective_techdisk_pickup)
+	else
+		UnregisterSignal(tech_disk, COMSIG_ITEM_PICKUP)
+
+/datum/objective/smart/download/proc/on_objective_techdisk_pickup(datum/tech_disk, mob/taker)
+	SIGNAL_HANDLER
+
+	if(taker == owner.current)
+		complete_objective()
+		UnregisterSignal(tech_disk, COMSIG_ITEM_PICKUP)
+		RegisterSignal(tech_disk, COMSIG_ITEM_DROPPED, .proc/on_objective_techdisk_dropped)
+	else
+		uncomplete_objective()
+
+/datum/objective/smart/download/proc/on_objective_techdisk_dropped(obj/item/disk/tech_disk/tech_disk, mob/dropper)
+	SIGNAL_HANDLER
+
+	uncomplete_objective()
+	UnregisterSignal(tech_disk, COMSIG_ITEM_DROPPED)
+	if(tech_disk.stored_research.researched_nodes.len >= target_amount)
+		RegisterSignal(tech_disk, COMSIG_ITEM_PICKUP, .proc/on_objective_techdisk_pickup)
+
+/datum/objective/smart/steal
+	name = "steal"
+	var/datum/objective_item/targetinfo = null //Save the chosen item datum so we can access it later.
+	var/obj/item/steal_target = null //Needed for custom objectives (they're just items, not datums).
+	martyr_compatible = FALSE
+	black_telecrystal_reward = 4
+
+/datum/objective/smart/steal/get_target()
+	return steal_target
+
+/datum/objective/smart/steal/New()
+	..()
+	if(!GLOB.possible_items.len)//Only need to fill the list when it's needed.
+		for(var/type in subtypesof(/datum/objective_item/steal))
+			new type
+
+/datum/objective/smart/steal/find_target(dupe_search_range)
+	var/list/datum/mind/owners = get_owners()
+	if(!dupe_search_range)
+		dupe_search_range = get_owners()
+	var/approved_targets = list()
+	check_items:
```

Comment:
the forbidden syntax

Reply:
i always forget this works until i see it in old code

Reply:
named continues and breaks are good actually

Reply:
if its in rust its good

Reply:
yeah but this is dream "datum var access" maker